src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/SubjectAccessRequestService.kt

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.service
+
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.incentivesapi.dto.ReviewType
+import uk.gov.justice.digital.hmpps.incentivesapi.jpa.repository.IncentiveReviewRepository
+import uk.gov.justice.digital.hmpps.incentivesapi.jpa.repository.NextReviewDateRepository
+import uk.gov.justice.hmpps.kotlin.sar.HmppsPrisonSubjectAccessRequestReactiveService
+import uk.gov.justice.hmpps.kotlin.sar.HmppsSubjectAccessRequestContent
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Service
+class SubjectAccessRequestService(
+  private val incentiveReviewRepository: IncentiveReviewRepository,
+  private val nextReviewRepository: NextReviewDateRepository,
+) : HmppsPrisonSubjectAccessRequestReactiveService {
+
+  override suspend fun getPrisonContentFor(
+    prn: String,
+    fromDate: LocalDate?,
+    toDate: LocalDate?,
+  ): HmppsSubjectAccessRequestContent? {
+    val reviews = incentiveReviewRepository.findAllByPrisonerNumberOrderByReviewTimeDesc(prn)
```

Comment:
i think it’d be better to use the `PrisonerIncentiveReviewService.getPrisonerIncentiveHistory` service method which already sorts out calculating the next review date if it is not available in the repository…

Reply:
Yes true, was just trying to get the raw data but agree